src/frontend/src/catalog/table_catalog.rs

```diff
@@ -487,6 +490,29 @@ impl TableCatalog {
     pub fn has_generated_column(&self) -> bool {
         self.columns.iter().any(|c| c.is_generated())
     }
+
+    pub fn get_output_indices(&self) -> &[usize] {
+        &self.output_indices
+    }
+
+    /// Creates a new `TableCatalog`.
+    /// It should not clone OR mutate the original table catalog.
+    /// Instead it should move it, so the old definition will not be reused,
+    /// leading to inconsistencies.
+    pub fn into_replicated(self, output_indices: Vec<usize>) -> Self {
+        // FIXME: Support value indices for ColumnAwareSerde.
+        // let mut output_value_indices = vec![];
+        // for output_idx in &output_indices {
+        //     if self.value_indices.contains(output_idx) {
+        //         output_value_indices.push(*output_idx);
+        //     }
+        // }
+        Self {
+            // value_indices: output_value_indices,
+            output_indices,
+            ..self
+        }
+    }
```

Comment:
We won't do replication in this PR, since that still requires several changes. See PR description for more details.